external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpoutConfig.java

```diff
@@ -115,26 +115,30 @@ public KafkaSpoutConfig(Builder<K, V> builder) {
     }
 
     /**
-     * The offset used by the Kafka spout in the first poll to Kafka broker. The choice of this parameter will affect the number of consumer
-     * records returned in the first poll. By default this parameter is set to UNCOMMITTED_EARLIEST. <br/><br/>
+     * Defines the offset used by the {@link KafkaSpout} in the first poll to Kafka broker. The choice of this parameter will affect
+     * the number of consumer records returned in the first poll. By default this parameter is set to UNCOMMITTED_EARLIEST. <br/><br/>
      * The allowed values are EARLIEST, LATEST, UNCOMMITTED_EARLIEST, UNCOMMITTED_LATEST. <br/>
      * <ul>
-     * <li>EARLIEST means that the kafka spout polls records starting in the first offset of the partition, regardless of previous
-     * commits</li>
-     * <li>LATEST means that the kafka spout polls records with offsets greater than the last offset in the partition, regardless of
-     * previous commits</li>
+     * <li>EARLIEST when the topology is first deployed the kafka spout polls records starting in the first offset of the
```

Comment:
OK, however this statement is correct because if a topology is deployed again it is a new topology because it has a new id and creates new object instances.